lib/nrf_modem_lib/nrf91_sockets.c

```diff
@@ -540,6 +533,15 @@ static ssize_t nrf91_socket_offload_recvfrom(void *obj, void *buf, size_t len,
 		retval = nrf_recvfrom(ctx->nrf_fd, buf, len, flags,
 				      NULL, NULL);
 	} else {
+		/* Dereference only if not NULL,
+		 * also prevent segfault later on
+		 */
+		if (fromlen == NULL) {
+			errno = EINVAL;
+			retval = -1;
+			goto exit;
```

Comment:
We could get rid of this check by including it in the previous `if (from == NULL)` test, e.g.
`if (from == NULL || fromlen == NULL)`

So we have two branches:
- one or both are NULL
- both are non-NULL